src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -102,6 +102,14 @@ public Task<ImmutableArray<MappedSpanResult>> MapSpansAsync(Document document, I
 
                 return Task.FromResult(mappedResult);
             }
+
+            public Task<ImmutableArray<(string mappedFilePath, TextChange mappedTextChange)>> GetTextChangesAsync(
+                Document oldDocument,
+                Document newDocument,
+                CancellationToken cancellationToken)
+            {
+                return Task.FromResult(ImmutableArray<(string, TextChange)>.Empty);
```

Comment:
i think the default impl shoudl either be to throw (so that we know we have to impl this), or call into oldDocument.GetTextChanges(newDocument)